packages/nimble-components/src/spinner/index.ts

```diff
@@ -0,0 +1,44 @@
+import { attr } from '@microsoft/fast-element';
+import {
+    applyMixins,
+    ARIAGlobalStatesAndProperties,
+    DesignSystem,
+    FoundationElement
+} from '@microsoft/fast-foundation';
+import { styles } from './styles';
+import { template } from './template';
+import type { SpinnerSize } from './types';
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-spinner': Spinner;
+    }
+}
+
+/**
+ * Spinner
+ */
+export class Spinner extends FoundationElement {
+    /**
+     * The size of the spinner (preset size options chosen by designers).
+     * If unset, the spinner will scale based on its width/height styles,
+     * or will be size 32x32 if not specified.
+     *
+     * @public
+     * @remarks
+     * HTML Attribute: size
+     */
+    @attr
+    public size: SpinnerSize | undefined;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface Spinner extends ARIAGlobalStatesAndProperties {}
+applyMixins(Spinner, ARIAGlobalStatesAndProperties);
```

Comment:
I'm not familiar with ARIAGlobalStatesAndProperties. What does this accomplish? Could we add a note to our contributing docs somewhere describing when it should be used? Are there any unit test patterns we should use to verify that this was done correctly?

Reply:
I think I was looking at the drawer or dialog as a starting point (as they also just derive from FoundationElement). Actually not sure if I should have it here though (Icon also derives from FoundationElement but doesn't have it).

@mollykreis do you have more info on what cases we should have these `ARIAGlobalStatesAndProperties` lines?

Reply:
I looked at the FAST progress indicators and they don't have this, so omitted it for now, we can add it later on if needed.